### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by yhaliaw**
Applicable spec: NA

### Overview

- Add template for future pull requests.
- Generate docs with `src-docs`
- Publish charm on push to main

### Rationale

- Simplify the process of PR creation.
- Simplify docs generation.
- Automate publish of charm.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by jdkandersson — DISMISSED** [review]

File: tox.ini [hunk 1]
```diff
@@ -111,3 +111,15 @@ deps =
...
+allowlist_externals=sh
+setenv =
+    PYTHONPATH=src
+description = Generate documentation for src
+deps =
+    lazydocs
+    -r{toxinidir}/requirements.txt
+commands =
+    ; can't run lazydocs directly due to needing to run it on src/* which produces an invocation error in tox
+    sh generate-src-docs.sh
```
Comment (jdkandersson) [inline tox.ini:125]: Final new line?

File: .github/workflows/publish_charm.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Publish to edge
```
Comment (arturo-seijas) [inline .github/workflows/publish_charm.yaml:1]: Next time, can you file unrelated changes like this with another PR?

**Review by arturo-seijas — DISMISSED** [review]
LGTM

File: .github/workflows/publish_charm.yaml (hunk 2 above)
Comment (yhaliaw) [reply .github/workflows/publish_charm.yaml:1]: Ok

**Review by arturo-seijas — DISMISSED** [review]

**Comment by yhaliaw** [comment]
The removal of `from __future__ import annotations` is needed as lazy-docs crashes with it.
The line can be removed anyways, as we are only supporting jammy (python 3.10) and up now.

**Review by jdkandersson — APPROVED** [review]

**Review by cbartz — APPROVED** [review]
